specutils/analysis/uncertainty.py

```diff
@@ -186,3 +187,26 @@ def _snr_derived(spectrum, region=None):
         return signal / noise
     else:
         return 0.0
+
+
+def _convert_uncertainty(uncertainty, to_class):
```

Comment:
This should really be in `astropy.nddata`.  And hopefully will be in 5.1 with stropy/astropy#12057 !  We should change this to use that if the version is > 5.0.x, and fall back on this implementation (or a simple copy-paste of 12057 into here?)  if not.  That way we won't be forced to require Astropy 5.1 which isn't even out yet.

Reply:
@eteq I'm revisiting this now - are there good reasons to keep `_convert_uncertainty` rather than just bumping the `astropy` requirement to 5.1 in the next `specutils` release and using the `represent_as` machinery now available on the uncertainty objects?